fix(editor): restrict servers paths for  `oxc.path.server`

Sysix opened the pull request:
A project can have a custom `settings.json` which paths like:

```
{
  "oxc.path.server": "/usr/known-other-library",
  "oxc.path.server": "../../outside-of-the-trusted-workspace",
  "oxc.path.server": "oxc_language_server|cat /etc/passwd"
}
```

Even if the user has trusted the workspace in #13734, "unexpected" changes in a project can happen :) 
Absolute files are allowed. For that, the binary should have `oxc_language_server` in the name, so the extension will not call another binary

Sysix commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13740?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13740** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13740?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13740?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#13738** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13738?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#13734** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13734?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13740).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented:
### Merge activity

* **Sep 15, 1:23 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
fix(editor): restrict servers paths for  `oxc.path.server` (#13740)

A project can have a custom `settings.json` which paths like:

```
{
  "oxc.path.server": "/usr/known-other-library",
  "oxc.path.server": "../../outside-of-the-trusted-workspace",
  "oxc.path.server": "oxc_language_server|cat /etc/passwd"
}
```

Even if the user has trusted the workspace in #13734, "unexpected" changes in a project can happen :)
Absolute files are allowed. For that, the binary should have `oxc_language_server` in the name, so the extension will not call another binary
